tests/deployment/vlan_networks.py

```diff
@@ -166,14 +200,31 @@ def delete_networks():
             # interference. If we were to hang on to references to database
             # objects across such calls however, things could get harry.
             all_attachments = []
+            net = namedtuple('net', 'node nic network channel')
             for node in nodes[:2]:
                 attachments = model.NetworkAttachment.query \
                     .filter_by(nic=node.nics[0]).all()
                 for attachment in attachments:
-                    all_attachments.append((node.label,
-                                            node.nics[0].label,
-                                            attachment.network.label))
+                    all_attachments.append(
+                                        net(node=node.label,
+                                            nic=node.nics[0].label,
+                                            network=attachment.network.label,
+                                            channel=attachment.channel))
+
+            switch = nodes[0].nics[0].port.owner
+            # in some switches, the native network can only be disconnected
+            # after we remove all tagged networks first. The following checks
+            # for that and rearranges the networks (all_attachments) such that
+            # tagged networks are removed first.
+
+            if 'nativeless-trunk-mode' not in switch.get_capabilities():
+                # sort by channel; so vlan/<integer> comes before vlan/native
+                all_attachments = sorted(all_attachments,
+                                         key=lambda net: net.channel)
```

Comment:
I'm still a bit bothered by the sorting thing; it's serendipitous that digits happen to come before letters in ascii/unicode, but that's a weird bit of knowledge to lean on for this, since it's so conceptually orthoginal. I'd be okay with it if we just improved the comment to detail exactly why this works.